benchmarks/__main__.py

```diff
@@ -47,31 +49,35 @@ def load_samples(zip_url, inputs_dir, exclude_attachments):
     return sample_emails
 
 
-def print_error(stage, compressor, serializer, ex):
-    print('Error during {}-phase in {}+{}: {}'.format(
+def print_error(stage, compressor, serializer, encryptor, ex):
+    print('Error during {}-phase in {}+{}+{}: {}'.format(
         stage,
         pretty_extension(compressor.extension),
         pretty_extension(serializer.extension),
+        pretty_extension(encryptor.extension),
         ex,
     ), file=stderr)
 
 
 def run_benchmarks(emails, results_dir, incremental):
     makedirs(results_dir, exist_ok=True)
 
-    for compressor, serializer in product(compressors(), serializers()):
-        outpath = join(results_dir, 'emails{}{}'.format(
-            serializer.extension, compressor.extension))
+    for compressor, serializer, encryptor in product(compressors(), serializers(), encryptors()):
+        outpath = join(results_dir, 'emails{}{}{}'.format(
+            serializer.extension, compressor.extension, encryptor.extension))
 
         if incremental and isfile(outpath):
             continue
 
         try:
             with Timer.timeit() as write_timer:
-                with compressor.open_write(outpath) as fobj:
-                    serializer.serialize(iter(emails), fobj)
+                with open(outpath, "wb") as fobj:
+                    with encryptor.encrypt(fobj) as enc:
```

Comment:
I'm having a hard time figuring out how the encryptor works so I tried adding a unit test like this:

```patch
diff --git a/benchmarks/tests.py b/benchmarks/tests.py
index 653df57..4bb6ec7 100644
--- a/benchmarks/tests.py
+++ b/benchmarks/tests.py
@@ -6,15 +6,31 @@ from tempfile import mkstemp
 from unittest import TestCase
 
 from benchmarks.compression import get_all as compressors
+from benchmarks.encryption import get_all as encryptors
 from benchmarks.serialization import get_all as serializers
 
 
-class CompressionTests(TestCase):
+class TempfilesTestCase(TestCase):
+    def setUp(self):
+        self.temp_paths = []
+
+    def tearDown(self):
+        for temp_path in self.temp_paths:
+            remove(temp_path)
+
+    def given_tempfile(self, extension):
+        fd, temp_path = mkstemp(suffix=extension)
+        close(fd)
+        self.temp_paths.append(temp_path)
+        return temp_path
+
+
+class CompressionTests(TempfilesTestCase):
     def test_roundtrip(self):
         for compressor in compressors():
             with self.subTest(compressor=compressor):
                 expected = b'test content'
-                path = self.given_tempfile(compressor)
+                path = self.given_tempfile(compressor.extension)
 
                 with compressor.open_write(path) as compressed:
                     compressed.write(expected)
@@ -24,19 +40,6 @@ class CompressionTests(TestCase):
 
                 self.assertEqual(actual, expected)
 
-    def setUp(self):
-        self.temp_paths = []
-
-    def tearDown(self):
-        for temp_path in self.temp_paths:
-            remove(temp_path)
-
-    def given_tempfile(self, compressor):
-        fd, temp_path = mkstemp(suffix=compressor.extension)
-        close(fd)
-        self.temp_paths.append(temp_path)
-        return temp_path
-
 
 class SerializationTests(TestCase):
     def test_roundtrip(self):
@@ -57,6 +60,26 @@ class SerializationTests(TestCase):
                 self.assertListEqual(actual, expected)
 
 
+class EncryptionTests(TempfilesTestCase):
+    def test_roundtrip(self):
+        for encryptor in encryptors():
+            with self.subTest(encryptor=encryptor):
+                expected = b'some bytes'
+
+                path = self.given_tempfile(encryptor.extension)
+
+                with open(path, 'wb') as fobj:
+                    with encryptor.encrypt(fobj) as encrypted:
+                        encrypted.write(expected)
+
+                with open(path, 'rb') as fobj:
+                    self.assertNotEqual(fobj.read(), expected)
+
+                with open(path, 'rb') as fobj:
+                    with encryptor.deserialize(fobj) as decrypted:
+                        self.assertEqual(decrypted.read(), expected)
+
+
 if __name__ == '__main__':
     from unittest import main
     main()
```

However, this test fails since `fobj.read()` is the same as `expected` which I find surprising since the underlying file should have been encrypted, right? What am I doing wrong? Would it be possible for you to add a unit test that verifies the encryption behavior? Thanks in advance!